lib/cdo/contact_rollups/v2/pardot.rb

```diff
@@ -0,0 +1,45 @@
+require_relative 'pardot_helpers'
+
+class PardotV2
+  extend PardotHelpers
+
+  PARDOT_API_V4_BASE = "https://pi.pardot.com/api/prospect/version/4".freeze
+  PARDOT_PROSPECT_QUERY_URL = "#{PARDOT_API_V4_BASE}/do/query".freeze
+
+  # Retrieves new email-id mappings from Pardot
+  # @param [Integer] last_id
+  # @return [Array] an array of hash {email, id}
+  def self.retrieve_new_ids(last_id = 0)
```

Comment:
This optional param allows us to have more control when we test or debug downstream syncing (without saving to db) in production.

Reply:
I'm not suggesting the param is a bad idea, just that it should be required (since it is set to 0 in `ContactRollupsPardotMemory`). Might also prevent us from accidentally fetching all contacts from Pardot 😄

Reply:
Ah good point. Will fix.